### PR TITLE
Add GitHub Action for automated testing

### PR DESCRIPTION
**Pull request by frangio**
Runs tests on commits and pull requests.

### CODE REVIEW

**Comment by haltman-at** [comment]
(I don't know GitHub actions well enough to comment on this.  Don't know whether @marcoscaceres or @pospi might have anything to say about it, or whether maybe it makes more sense to just go ahead and merge this without someone else reviewing... like I said, GHA is not my area!)

**Comment by gnidan** [comment]
Should be able to test this by branching off into a new PR and adding an intentional failure

**Comment by frangio** [comment]
Example commit with a failing test: https://github.com/highlightjs/highlightjs-solidity/commit/628ebbd7af83984f99fa9d4b3fc039fbd08c399e

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+  - pull_request
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v1
+    - uses: actions/setup-node@v1
+      with:
+        node-version: 10.x
```
Comment (marcoscaceres) [inline .github/workflows/test.yml]: I think 12 is lts.

**Pull request merged**